US: #2029018 Send custom properties to ALM Octane on Job completion

bbokhonko-mf opened the pull request:
**Can't be merged now due to missing new version of CI SDK**

Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!

- [x] Proper pull request title - concise and clear for others.
- [ ] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [ ] Proper Jira ticket - Number, Link in pull request description.
- [x] The PR can is merged on your machine without any conflicts.
- [x] The PR can is built on your machine without any (new) warnings.
- [ ] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [ ] Add unit tests with new features.
- [ ] If you added any dependency to the POM - Please update grount

bbokhonko-mf committed:
US: #2029018 Send custom properties to ALM Octane on Job completion

bbokhonko-mf committed:
US: #2029018 Send custom properties to ALM Octane on Job completion (Refactoring)

bbokhonko-mf committed:
US #2048033: Extend polling to send/receive custom properties

nissimshitrit commented on src/main/java/com/microfocus/application/automation/tools/octane/events/AbstractBuildListenerOctaneImpl.java:
```diff
@@ -126,7 +128,8 @@ private void publishFinishEvent(AbstractBuild build) {
 					.setDuration(build.getDuration())
-					.setTestResultExpected(hasTests);
+					.setTestResultExpected(hasTests)
+					.setEnvironmentOutputtedParameters(getOutputEnvironmentParams(build));
```
Can you please use the helper inside the set method, it more readable like other setters instead of adding it on import

bbokhonko-mf replied:
done

nissimshitrit commented on src/main/java/com/microfocus/application/automation/tools/octane/model/processors/projects/AbstractProjectProcessor.java:
```diff
@@ -209,6 +211,7 @@ public CIBuildStatusInfo getBuildStatus(String paramName, String paramValue) {
 							} else {
 								status.setBuildStatus(CIBuildStatus.FINISHED);
 								status.setResult(BuildHandlerUtils.translateRunResult(aBuild));
+								status.setEnvironmentOutputtedParameters(getOutputEnvironmentParams(aBuild));
```
Same for here, please use the helper inside the set method, it more readable like other setters instead of adding it on import

bbokhonko-mf replied:
done

nissimshitrit commented on src/main/java/com/microfocus/application/automation/tools/octane/events/OutputEnvironmentParametersHelper.java:
```diff
@@ -0,0 +1,80 @@
+		EnvVars environment = null;
+		try {
+			environment = run.getEnvironment(null);
+		} catch (IOException | InterruptedException ignored) {
```
Can you please add to plugin about error here, no need all exception log , can put general message and add exception message

bbokhonko-mf replied:
added, please verify the message

nissimshitrit reviewed (changes requested):
I see you are handling one type of finish events, do you need to handle on finish event of Pipeline jobs (Workflow) you can find plugin code handling finish event of Pipeline jobs here WorkflowListenerOctaneImpl.java:157
In addition in order that plugin will compile with changes done on SDK you should add latest SDK version to plugin pom, the last version I created is 2.7.8.0

bbokhonko-mf committed:
US #2048033: Refactoring

bbokhonko-mf commented:
> I see you are handling one type of finish events, do you need to handle on finish event of Pipeline jobs (Workflow) you can find plugin code handling finish event of Pipeline jobs here WorkflowListenerOctaneImpl.java:157 In addition in order that plugin will compile with changes done on SDK you should add latest SDK version to plugin pom, the last version I created is 2.7.8.0

Thank you for "Pipeline jobs" suggestion, I missed it. Added it on last commit.

Also updated SDK version, and now checks are passed
